File: app/src/main/java/com/arad/care4pets/PetAdapter.java
package com.arad.care4pets;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class PetAdapter extends RecyclerView.Adapter<PetAdapter.PetViewHolder> {

    public interface OnItemClickListener {
        void onItemClick(Pet pet);
    }

    private List<Pet> pets;
    private final OnItemClickListener onItemClickListener;

    public PetAdapter(List<Pet> pets, OnItemClickListener onItemClickListener) {
        this.pets = pets;
        this.onItemClickListener = onItemClickListener;
    }

    public void setPets(List<Pet> pets) {
        this.pets = pets;
        notifyDataSetChanged();
    }

    public static class PetViewHolder extends RecyclerView.ViewHolder {
        TextView tvName;
        TextView tvSpecies;

        public PetViewHolder(@NonNull View itemView) {
            super(itemView);
            tvName = itemView.findViewById(R.id.tvPetName);
            tvSpecies = itemView.findViewById(R.id.tvPetSpecies);
        }
    }

    @NonNull
    @Override
    public PetViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
            .inflate(R.layout.item_pet, parent, false);
        return new PetViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull PetViewHolder holder, int position) {
        Pet pet = pets.get(position);
        holder.tvName.setText(pet.getName());
        String speciesAge = pet.getSpecies() + ", " + pet.getAge() + " years";
        holder.tvSpecies.setText(speciesAge);

        holder.itemView.setOnClickListener(v -> onItemClickListener.onItemClick(pet));
    }

    @Override
    public int getItemCount() {
        return pets != null ? pets.size() : 0;
    }
}

File: app/src/main/java/com/arad/care4pets/PetAdapter.kt
package com.arad.care4pets

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class PetAdapter(
    pets: List<Pet>,
    private val onItemClick: (Pet) -> Unit
) : RecyclerView.Adapter<PetAdapter.PetViewHolder>() {

    var pets: List<Pet> = pets
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    inner class PetViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvName: TextView = itemView.findViewById(R.id.tvPetName)
        val tvSpecies: TextView = itemView.findViewById(R.id.tvPetSpecies)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PetViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_pet, parent, false)
        return PetViewHolder(view)
    }

    override fun onBindViewHolder(holder: PetViewHolder, position: Int) {
        val pet = pets[position]
        holder.tvName.text = pet.name
        holder.tvSpecies.text = "${pet.species}, ${pet.age} years"
        holder.itemView.setOnClickListener { onItemClick(pet) }
    }

    override fun getItemCount(): Int = pets.size
}
